### DISCUSSION TITLE
Infinite loop

### DISCUSSION THREAD

**Discussion by zhex900** [directus/directus#14088] [Feature Requests]
### Describe the Bug

I accidentally created an infinite loop in the flow. Now the project is crashed. 

Flow should have some kind of protection against this.

### To Reproduce

1. Trigger: update item (post)
2. Log it
3. Update item (post)

Step 3 will trigger step 1. Now it never stops. 

### What version of Directus are you using?

v9.12.2 latest

### What version of Node.js are you using?

Paas

### What database are you using?

Paas

### What browser are you using?

firebox

### How are you deploying Directus?

Paas

**Comment by br41nslug** [comment]
This should be prevented by default with this PR #13121 but seems it is not released yet.

**Reply by nervusdm** [reply]
Waiting for it. I just had this same problem.

**Reply by br41nslug** [reply]
It should be available already @nervusdm 

![image](https://user-images.githubusercontent.com/9389634/181517612-45bb2f4a-b6a4-47ef-8cec-8ba9107c553c.png)

**Reply by nervusdm** [reply]
All right, thanks.

**Comment by tatthien** [comment]
Hi @br41nslug 
I believe there's still an infinite loop bug here. When you uncheck `Enabled`  (or leave it unchecked by default), the option `emitEvents` is `undefined`. It's not `false`. 
The following code check if `emitEvents  !== false` then run the emit event. `undefined !== false`  => `true` and the event is still emitted.
<img width="1004" alt="Screen Shot 2022-08-03 at 11 45 08" src="https://user-images.githubusercontent.com/72242664/182527187-3c812104-1323-4549-88f5-8bbf64a5431d.png">

**Reply by azrikahar** [reply]
This was recently fixed in #14741 and will be included in the [next release](https://github.com/directus/directus/milestone/117) 👍

**Reply by tatthien** [reply]
Great! Thanks

**Comment by rijkvanzanten** [comment]
There's currently a way to manually prevent this through the Emit Events option in the flow trigger.
